### PR TITLE
Add default "resolvedIf" function for tasks

### PR DESCRIPTION
**Pull request by binokaryg**
# Description

Adds a default `resolvedIf` function for tasks.

medic/medic-conf#107

## defaultResolvedIf
- Uses [Utils.isFormSubmittedInWindow](https://github.com/medic/medic-nootils/blob/4339b25201bfbf51eef9179c8b7e0e92734a0d48/src/nootils.js#L57) to check if the resolving form has been submitted within a time period:
    - For contact-based tasks, the period is the same as the task window period
    - For report based tasks, the period is determined as follows:
        - start: start of the task window or 1 millisecond after reported date of the source report, whichever comes later
        - end: end of the task window
    - The resolving form is the task's first action form

## Usage
Options:
A. Skip `resolvedIf` section in the task definition, `defaultResolvedIf` will be used automatically
B. Using `this.defaultResolvedIf`:
1. Without specifying the parameters:
    ```
    resolvedIf: this.defaultResolvedIf,
    ```

2. Specifying parameters and optionally add other conditions:
    ```
    resolvedIf: function (contact, report, event, dueDate) {
      return this.defaultResolvedIf(contact, report, event, dueDate) && ...
    },
    ```
3. Specifying the resolving form type if it is different from the default (form type specified in the first action of the task):
    ```
    resolvedIf: function (contact, report, event, dueDate) {
      return this.defaultResolvedIf(contact, report, event, dueDate, 'delivery');
    },
    ```
# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: test/run-nools-lib.js [hunk 1]
```diff
@@ -19,6 +19,28 @@ const runNoolsLib = ({ c, targets, tasks }) => {
...
+              if (!count ||
+                 (count && report.fields && report.fields.follow_up_count > count)) {
+                result = true;
+              }
+            }
+          }
+        });
+        return result;
+      },
+      defaultResolvedIf: function (contact, report, event, dueDate, resolvingForm) {
```
Comment (kennsippell) [inline test/run-nools-lib.js]: weird that this is duplicated in test code...

File: test/nools/task-emitter.spec.js [hunk 2]
```diff
@@ -287,6 +306,25 @@ describe('task-emitter', () => {
         });
       });
 
+      it('resolvedIf is not required', () => {
```
Comment (kennsippell) [inline test/nools/task-emitter.spec.js:291]: shouldn't have two tests with the same name. what's the difference?

**Review by kennsippell — APPROVED** [review]
u get it

File: test/run-nools-lib.js (hunk 1 above)
Comment (binokaryg) [reply test/run-nools-lib.js]: removed

File: test/nools/task-emitter.spec.js (hunk 2 above)
Comment (binokaryg) [reply test/nools/task-emitter.spec.js:291]: One was under "person based task" and another under "report based task", but that might be unnecessary, so only keeping one

**Pull request merged**